scripts/manageacl.py

```diff
@@ -0,0 +1,176 @@
+import argparse
+import asyncio
+import logging
+
+from azure.core.credentials import AzureKeyCredential
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.identity import AzureDeveloperCliCredential
+from azure.search.documents.aio import SearchClient
+from azure.search.documents.indexes.aio import SearchIndexClient
+from azure.search.documents.indexes.models import SearchFieldDataType, SimpleField
+
+
+class ManageAcl:
+    """
+    Manually enable document level access control on a search index and manually set access control values using the [manageacl.ps1](./scripts/manageacl.ps1) script.
+    """
+
+    def __init__(
+        self,
+        service_name: str,
+        index_name: str,
+        document: str,
+        acl_action: str,
+        acl_type: str,
+        acl: str,
+        credentials: AsyncTokenCredential | AzureKeyCredential,
+    ):
+        """
+        Initializes the command
+
+        Parameters
+        ----------
+        service_name
+            Name of the Azure Search service
+        index_name
+            Name of the Azure Search index
+        document
+            File path of the document to manage acls for
+        acl_action
+            Action to take regarding the index or document. Valid values include enable_acls (turn acls on for the entire index), view (print acls for the document), remove_all (remove all acls), remove (remove a specific acl), or add (add a specific acl)
+        acl_type
+            Type of acls to manage. Valid values include groups or oids.
+        acl
+            The actual value of the acl, if the acl action is add or remove
+        credentials
+            Credentials for the azure search service
+        """
+        self.service_name = service_name
+        self.index_name = index_name
+        self.credentials = credentials
+        self.document = document
+        self.acl_action = acl_action
+        self.acl_type = acl_type
+        self.acl = acl
+
+    async def run(self):
+        endpoint = f"https://{self.service_name}.search.windows.net"
+        async with SearchClient(
+            endpoint=endpoint, credential=self.credentials, index_name=self.index_name
+        ) as search_client, SearchIndexClient(endpoint=endpoint, credential=self.credentials) as search_index_client:
+            if self.acl_action == "enable_acls":
```

Comment:
I think it'd be nice design wise for this class to have enable_acls(), view_acls(), , remove_all_acls(), remove_acl() and add_acl(), and then run() can call those as needed. That way, if someone wanted to use this class via some other mechanism, like a UI or API, they could call the individual methods. It's also easier to test, I think.